ovirt_imageio/client/_ovirt.py

```diff
@@ -366,6 +446,43 @@ def finalize_transfer(con, transfer, disk, timeout=300):
              transfer.id, time.monotonic() - start)
 
 
+def _detect_disk_content_type(filename, image_format):
+    """
+    Detect disk content type
+
+    ISO format structure
+    ---------------------------------------------------------------------------
+    offset    type    value       comment
+    ---------------------------------------------------------------------------
+    0x0000                        system area (e.g. DOS/MBR boot sector)
+    0x8000    int8    0x01        primary volume descriptor type code
+    0x8001    strA    "CD001"     primary volume descriptor indentifier
+    0x8006    int8    0x01        primary volume desctptor version
+    0x8007            0x00        unused field
+
+    See https://wiki.osdev.org/ISO_9660#Overview_and_caveats for more info.
+
+    Returns:
+        ovirtsdk4.types.DiskContentType
+    """
+    if image_format == "raw":
+        with open(filename, "rb") as file:
+            file.seek(0x8000)
+            primary_volume_descriptor = file.read(8)
+        if primary_volume_descriptor == b"\x01CD001\x01\x00":
+            return types.DiskContentType.ISO
+    return types.DiskContentType.DATA
+
+
+def _wait_for_disk(con, disk_id):
+    disk_service = _disk_service(con, disk_id)
+    while True:
+        time.sleep(1)
+        disk = disk_service.get()
+        if disk.status == types.DiskStatus.OK:
```

Comment:
It is a good question. This is how it is handled by the `upload_disk` example, but it may be worth checking for error status or add a timeout option.

I would leave for another patch though, can be an enhancement.

Reply:
Ok, I'll add a fixed timeout now for the first deploy, and open an issue for error handling and (maybe) configurable timeout.

Reply:
https://github.com/oVirt/ovirt-imageio/issues/117